### PR TITLE
More intuitive Makefile targets for testing and packaging

### PR DESCRIPTION
**Pull request by conorsch**
## Status

Ready for review.

## Description of Changes

Dev-env only changes designed to make testing and packaging more straightforward. Includes some docs changes to track the new Makefile targets, and consolidated all the CI steps into a single `make test`. 

Trying to reduce the opportunities for error in both testing and packaging steps.

## Testing

Take the new targets for a spin and report any problems. Double-check that the docs are clear, and not confusing.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -15,14 +15,18 @@ clean: docker-clean ## Removes all build-related artifacts
...
+build-deb: docker-build ## Builds Sunder Debian packages for Linux
 	docker volume create fpf-sunder-node && \
 	docker run \
 		-v $(PWD):/sunder \
 		-v fpf-sunder-node:/sunder/node_modules \
 		sunder-build:latest
 
+.PHONY: build-dmg
+build-dmg: ## Builds Sunder DMG and app bundle for macOS (requires macOS)
+	npm run dist
```
Comment (emkll) [inline Makefile]: In macOs, `npm install` is required before `npm run dist` if it has not already run or if a user has invoked `make clean`. Perhaps something like `npm install && npm run dist` instead?

File: Makefile [hunk 2]
```diff
@@ -45,6 +49,21 @@ docs: docs-clean ## Runs livereload environment for local documentation editing
 # Spins up livereload environment for editing; blocks.
 	sphinx-autobuild docs/ docs/_build/html
 
+.PHONY: build-app
+build-app: ## Packages Electron app locally via webpack for development and testing
+	npm run build-app
+
+.PHONY: test-unit
```
Comment (emkll) [inline Makefile]: Same as above: for both `test-unit` and `test-2e2`, `npm install` needs to fetch and install libraries on first install or after `make clean`

**Comment by conorsch** [comment]
Great feedback, @emkll. Testing these changes on macOS currently, will tag you back for review once I push up changes.

**Comment by conorsch** [comment]
Hmm, `npm install` takes ~23s even when no changes are required. The `make test-unit` target takes only 3s, so I'm reticent to automatically chain the `npm install` and drastically slow down the test runs. I'll try checking for directory existence of `node_modules/` and only fire if necessary.

**Comment by conorsch** [comment]
@emkll Take a look at the changes—should address your concerns!

**Comment by conorsch** [comment]
Try `make clean && make test` as an example of the new automagic, implemented at your suggestion.

**Comment by emkll** [comment]
Thanks @conorsch, LGTM

**Pull request merged**